client/director.go

```diff
@@ -162,11 +158,11 @@ func queryDirector(ctx context.Context, verb, sourcePath, directorUrl string) (r
 		// If we get a 405 with a PROPFIND, the client will handle it
 		return
 	} else if resp.StatusCode != 307 {
```

Comment:
Oops. One thing we missed here is to check the response content type. If the content type is not "application/json", let's simply log `string(body)`